### ISSUE TITLE
Parallel execution using Mstest+reqnroll

### ISSUE THREAD

**Issue by sampath-ganesan** [reqnroll/Reqnroll#116]
### Reqnroll Version

1.0.1

### Which test runner are you using?

MSTest

### Test Runner Version Number

MsTest.TestAdapter 3.3.1

### .NET Implementation

.NET 8.0

### Test Execution Method

Visual Studio Test Explorer

### Issue Description

I am trying to run scenarios in parallel execution using mstest while doing that i am facing few issues,
1. The previous ScenarioStepContext was already disposed, The previous ScenarioContext was already disposed.
2. while running in 2 threads scenario step details are mixed up with other threads
3. step name is getting mixed up on the extend report, for some

### Steps to Reproduce

1. set up mstest project with reqnroll and add few basic scenarios and also use extend report 
2. set Parallelize via .runsettings
3. use StepContext.StepInfo

Note: I also tried

private readonly ScenarioContext scenarioContext;

public SpecflowHooks(ScenarioContext scenarioContext)
{
  this.scenarioContext = scenarioContext;
}

And

ScenarioContext.Current.ScenarioInfo

**Comment by ajeckmans** [comment]
Only feature level parallelization is supported for mstest. See also https://docs.reqnroll.net/latest/execution/parallel-execution.html

**Comment by ajeckmans** [comment]
Nothing that is concrete though it is definitely something that is kept in mind.

**Comment by sampath-ganesan** [comment]
thanks for the info

**Comment by gasparnagy** [comment]
@sampath-ganesan please also see my comment at #114. I recommend creating an idea discussion item for this.

I close this as by design, we can track this at the discussion item.

**Comment by sampath-ganesan** [comment]
i am able to open multiple browser with ThreadLocal and again that ScenarioContext is not supporting that ThreadLocal so its getting disposed after first scenario completes on that thread

**Comment by gasparnagy** [comment]
@sampath-ganesan please make a separate issue for that
